### PR TITLE
fix: use GitHub App token when authed with GitHub App

### PR DESCRIPTION
**Pull request by jmeridth**
# Pull Request

## Proposed Changes

Currently we are still trying to use the GH_TOKEN env var when making api/graphql calls even after authenticating with a GitHub App Installation. This change fixes that.

Also fixed a few other things while in here:

- [x] split authentication to auth.py file (like other actions)
- [x] fix arguments to the count_comments_per_user function

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer

- [x] Label as either `fix`, `documentation`, `enhancement`, `infrastructure`, `maintenance`, or `breaking`

### CODE REVIEW

File: issue_metrics.py [hunk 1]
```diff
@@ -175,9 +143,9 @@ def get_per_issue_metrics(
                 issue_with_metrics.mentor_activity = count_comments_per_user(
                     None,
                     issue,
-                    ignore_users,
                     None,
                     None,
+                    ignore_users,
```
Comment (zkoppert) [inline issue_metrics.py:148]: uh oh. This should have been caught by a test. Added one at 79a3995

Comment (jmeridth) [reply issue_metrics.py:148]: local linting caught it for me.  mypy ftw.  excellent idea to add tests.  thank you.

**Review by zkoppert — APPROVED** [review]
:shipit:

File: issue_metrics.py (hunk 1 above)
Comment (zkoppert) [reply issue_metrics.py:148]: Is mypy disabled in our super-linter? Wondering how this status check passed so we could merge.

**Pull request merged**